bundle/src/main/java/com/adobe/acs/commons/workflow/process/impl/WorkflowDelegationStep.java

```diff
@@ -0,0 +1,161 @@
+            value = "Workflow Delegation step",
+            description = "Calls configurable sub workflows"
+    )
+})
+public class WorkflowDelegationStep implements WorkflowProcess {
+	
+	
+	@Reference
+	ResourceResolverFactory rrf;
+
+	private static final Logger LOG = LoggerFactory.getLogger(WorkflowDelegationStep.class);
+	
+	// Under this property the workflow model is stored
+	private static final String WORKFLOW_PROPERTY_NAME = "workflowModelProperty";
+	
+	// A default workflow model can be provided as fallback
+	private static final String DEFAULT_WORKFLOW_MODEL = "defaultWorkflowModel";
+	
+	@Override
+	public void execute(WorkItem item, WorkflowSession session, MetaDataMap metadata)
+			throws WorkflowException {
+		
+		Map<String,String> args = getProcessArgsMap(metadata);
+		
+		String propertyName = args.get(WORKFLOW_PROPERTY_NAME);
+		if (StringUtils.isBlank(propertyName)) {
+			throw new WorkflowException ("propertyname not defined");
+		}
+		String defaultWorkflowModel = args.get(DEFAULT_WORKFLOW_MODEL);
+		
+		LOG.debug ("propertyName = {}, default workflow model = {}", 
+				new Object[]{propertyName,defaultWorkflowModel});
+		
+		ResourceResolver resolver = null;
+		WorkflowData wfData = item.getWorkflowData();
+		if (wfData.getPayloadType().equals("JCR_PATH")) {
+			String path = wfData.getPayload().toString();
+			
+			Session jcrSession = session.adaptTo(Session.class);
+			Map<String, Object> map = new HashMap<String, Object>();
+			map.put("user.jcr.session", jcrSession);
+			try {
+				resolver = rrf.getResourceResolver(map);
+			} catch (LoginException e) {
+				throw new WorkflowException (e);
+			}
+			
+			if (resolver != null) {
+				Resource wfResource = resolver.getResource(path);
+				InheritanceValueMap inheritance = new HierarchyNodeInheritanceValueMap(wfResource);
+				
+				String foundWorkflowModel = inheritance.getInherited(propertyName, String.class);
+				if (foundWorkflowModel != null) {
+					LOG.debug ("Using workflowmodel {}", foundWorkflowModel);
+				} else {
+					if (defaultWorkflowModel == null) {
+						throw new WorkflowException 
+							(String.format("Haven't found property %s on resource %s or higher, also defaultWorkflowModel not defined", 
+								new Object[]{propertyName,defaultWorkflowModel}));
+					} else {
+						foundWorkflowModel = defaultWorkflowModel;
+						LOG.debug("Falling back to workflowmodel {}", foundWorkflowModel);
+					}
+				} 
+				
+				
+				WorkflowModel newModel = session.getModel(foundWorkflowModel);
+				session.startWorkflow(newModel, wfData);
+				LOG.debug("Started workflow model {} with current payload", foundWorkflowModel);
+				
+			}
+				
+		}
+		
+	}
+
+    private Map<String, String> getProcessArgsMap(MetaDataMap metaDataMap) {
+        final Map<String, String> map = new LinkedHashMap<String, String>();
+        final String processArgs = metaDataMap.get("PROCESS_ARGS", "");
+        final String[] lines = StringUtils.split(processArgs, ",");
```

Comment:
Could use `return ParameterUtil.toMap(StringUtils.split(processArgs, ","), "=");` .. does the same thing.